packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -135,17 +135,19 @@ const ProductTile = (props) => {
                     }}
                 >
                     <IconButtonWithRegistration
+                        data-testid="wishlist-button"
                         aria-label={intl.formatMessage({
                             id: 'product_tile.assistive_msg.wishlist',
                             defaultMessage: 'Wishlist'
                         })}
                         icon={isFavourite ? <HeartSolidIcon /> : <HeartIcon />}
                         {...styles.favIcon}
-                        disabled={isFavouriteLoading}
                         onClick={async () => {
-                            setFavouriteLoading(true)
-                            await onFavouriteToggle(!isFavourite)
-                            setFavouriteLoading(false)
+                            if (!isFavouriteLoading.current) {
+                                isFavouriteLoading.current = true
+                                await onFavouriteToggle(!isFavourite)
+                                isFavouriteLoading.current = false
```

Comment:
Fix here is to useRef as useState is async and sometimes does not update fast enough to keep up with the changing loading values.

I removed the disabled property since we are no longer re-rendering when isFavouriteLoading changes.